### ISSUE TITLE
How to add and fine-tune negative images to a pre-trained model model in Yolov8?

### ISSUE THREAD

**Issue by kbratsy** [ultralytics/ultralytics#4761]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello, I have already trained a model in which the parameters in its _**data.yaml were nc:1 names: ['no_parking']**_. I then wanted to fine-tune this model to reduce FPs with unlabeled negative images. When retraining the model I ran into the error using only negative images.
First, I will share some input information to better explain my problem.

I added my negative images in train/images as below. Then I created the relevant blank .txts in the train/labels as there are negative examples of these images. I updated **_data.yaml (nc:2 names: ['no_parking','background'])_** .
![image](https://github.com/ultralytics/ultralytics/assets/34754399/804e725f-1c3d-41ee-a9a7-f1f3ff427fe2)

Then when I want to retrain my model with these negative images, as follows:
```
model = YOLO(r"..\train\weights\best.pt") 

results = model.train(data=r"..\data.yaml", epochs=100, imgsz=512,device=0, batch=8) 
```
I got an error. Error message, as follows:
**_ValueError: All labels empty in ..\train\labels.cache, can not start training without labels. See https://docs.ultralytics.com/yolov5/tutorials/train_custom_data_**

How can I use unlabeled negative images in yolov8 by fine-tuning the model? Where am I doing wrong? 

Thanks in advance.

**Comment by glenn-jocher** [comment]
@kbratsy hello,

Thanks for the detailed explanation of your challenge while working with YOLOv8. The issue you are experiencing occurs because YOLOv8 expects all images to have at least one object to detect. The error message suggests that all labels are empty because you have used blank `.txt` files for corresponding negative images. 

Adding negative examples to your training data is a good approach to reduce false positives. However, instead of adding a new class named 'background', you should keep the same number of classes (nc:1) and same classes names, ['no_parking']. In YOLO, the background is usually implicitly defined by what is not part of the bounding boxes.

When you plan to incorporate the negative images (images with no objects of interest), simply include these images in your training set and provide the corresponding blank `.txt` files in the labels directory, just like you have already done. Make sure the `data.yaml` remains unchanged with the single class like before ('no_parking'). However, remember not to solely train on these negative images; include a mix of positive and negative images. The model will then learn to recognize what isn't a no-parking sign, alongside what is.

Hope this helps you proceed with your model fine-tuning and thank you for using YOLOv8! Let me know if you have any other questions.

Best regards,
Glenn

**Comment by kbratsy** [comment]
Hi @glenn-jocher, thank you for your answer. I did what you said (nc:1) and ['no_parking']. I already have related .txt files for existing negative images which are empty. But I am still getting the same error. I just want to make a note that I want to fine-tune an already trained model by only training negative images in the model, not training a model from scratch. Is there something I missed from what you said, which I will try to show in more detail below?
![image](https://github.com/ultralytics/ultralytics/assets/34754399/1b2c7dd8-f82a-4c6d-8c33-e05ee0ab8173)

**Comment by kbratsy** [comment]
Hello @glenn-jocher , I understand that we do not fine-tune the model only with background images. Thanks for your answers.

**Comment by glenn-jocher** [comment]
@kbratsy, you're welcome and thank you for your understanding.

That's correct, fine-tuning a model effectively requires a mix of positive and negative samples to maintain the balance in learning. This will ensure that the model continues to improve its ability to detect the 'no_parking' sign, while at the same time, reducing the false positives by learning from the negative examples as well. 

If you have any more questions or need further assistance, please don't hesitate to ask. We are here to help you!

Happy coding with YOLOv8!

Best,
Glenn
